### PR TITLE
layers: Block Texel View VU 01583

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Touches test from #2214
Not fixing, but related to #1495 

Adds support for `VUID-VkImageViewCreateInfo-image-01583`

confirmed that "size-compatible" refers to the `FormatElementSize`

(CTS has a test that goes from `VK_FORMAT_BC3_UNORM_BLOCK` to `VK_FORMAT_R32G32B32A32_UINT` which both are a 128-bit format)

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -5940,14 +5940,26 @@ bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageVi
...
                              "vkCreateImageView(): Image was created with VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT bit and "
                              "non-compressed format (%s), but pCreateInfo->viewType is VK_IMAGE_VIEW_TYPE_3D.",
                              string_VkFormat(image_format));
+            }
+
+            const bool class_compatible = FormatCompatibilityClass(view_format) == FormatCompatibilityClass(image_format);
+            // "uncompressed format that is size-compatible" so if compressed, same as not being compatible
+            const bool size_compatible =
+                FormatIsCompressed(view_format) ? false : FormatElementSize(view_format) == FormatElementSize(image_format);
+            if (!class_compatible && !size_compatible) {
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:5956]: For my own clarification: it _is_ valid for the view and image formats to be class compatible but _not_ size compatible (and vice-versa)?

**Review by ncesario-lunarg — APPROVED** [review]
One clarifying question on my part, but LGTM!

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:5956]: > If image was created with the VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT flag, **format** must be compatible with, or must be an uncompressed format that is size-compatible with, the format used to create image

so you can strip it out to say

> **format** must be compatible with, ... , the format used to create image

----

do note that "class compatible" basically means "the same size"

if you look at `enum class FORMAT_COMPATIBILITY_CLASS` or the [table in the spec](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#resources-image-creation-limits) the classes are even named things like `8-bit`, `64-bit`, etc 

... so I guess one "could" argue the `class_compatible` check here might be redundant a bit, but rather not take a chance

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -5940,14 +5940,26 @@ bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageVi
                                  "but subresourcesRange.layerCount (%" PRIu32 ") is not 1.",
                                  pCreateInfo->subresourceRange.layerCount);
             }
-        }
 
-        if (image_flags & VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT && !FormatIsCompressed(view_format) &&
```
Comment (aitor-lunarg) [inline layers/buffer_validation.cpp:5945]: I'm not sure I understand why we are removing the check with image's creation flags. Same for below

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:5945]: if `if` statement above it, which I moved this into, already is doing the flag check

Comment (aitor-lunarg) [reply layers/buffer_validation.cpp:5945]: Thank you, I didn't notice the `}` removed above.

**Review by aitor-lunarg — APPROVED** [review]

**Pull request merged**